### ISSUE TITLE
[Bug] git aliases don't work

### ISSUE THREAD

**Issue by centreboard** [Azure/azure-devops-cli-extension#593]
**Describe the bug**
Trying to use git aliases results in 
> environment: line 0: exec: az: not found

**To Reproduce**
Azure Cli Version: 2.0.64
Azure-Devops extension version: 0.8.0

Steps to reproduce the behaviour:
```
> az devops configure --use-git-aliases yes
> git pr create
environment: line 0: exec: az: not found
```

**Expected behaviour**
`git pr` should have same behaviour as `az repos pr`
`git repo` should have same behaviour as `az repos`

**Additional info**
Reproduced on windows in cmd and pwsh.

from .gitconfig
```
[alias]
	pr = "!f() { exec az repos pr \"$@\"; }; f"
	repo = "!f() { exec az repos \"$@\"; }; f"
```

**Comment by Jaykul** [comment]
The `az` command in the alias needs a `.cmd` added to it on Windows.

**Comment by atbagga** [comment]
@centreboard @Jaykul  How did you install azure cli? Was it using the msi or just a pip install?

**Comment by centreboard** [comment]
@atbagga I used the msi

**Comment by qzdl** [comment]
The file is https://github.com/Azure/azure-devops-cli-extension/blob/master/azure-devops/azext_devops/dev/common/git.py

Encountering this was a bit of a pain

**Comment by gauravsaralMs** [comment]
@qzdl , @centreboard  is this happening when you are trying to use git from git bash?
Then this might be related to an issue in azure cli
https://github.com/Azure/azure-cli/issues/8560 

can you confirm if this happens from normal terminal also or just git bash..
Apologies for the questions but this will help us to root cause it correctly.

**Comment by centreboard** [comment]
I've been running from powershell/cmd prompts. 
`az repos pr create` works from them.
`git pr create` only works once I've edited the git config to use `az.cmd`

**Comment by gauravsaralMs** [comment]
looks like ultimately git alias is executed with git bash environment only
![image](https://user-images.githubusercontent.com/43203065/58274568-c8515400-7db0-11e9-89aa-696db1f3c255.png)

this proves that because cls is not recognized but clear works in cmd (which can't happen in cmd, but happens in bash)

**Comment by atbagga** [comment]
This is related to Azure CLI MSI not packaging/handling the az bash script. It does work for me on my windows machine since az bash script is picked from the python scripts folder. 

One quick fix for this could be to add windows specific hack in setting git aliases as suggested above and use az.cmd instead of az. 

I will explore if we can fix the Azure CLI MSI packaging and fix this properly else go with the workaround.

**Comment by qzdl** [comment]
> @qzdl , @centreboard is this happening when you are trying to use git from git bash?
> Then this might be related to an issue in azure cli
> [Azure/azure-cli#8560](https://github.com/Azure/azure-cli/issues/8560)
> 
> can you confirm if this happens from normal terminal also or just git bash..
> Apologies for the questions but this will help us to root cause it correctly.

This error occurs in powershell/git bash/cmd prompts.
I have no issue with other `az` commands in PS6

See a hacky fix in the below commit
https://github.com/qzdl/azure-devops-cli-extension/commit/595135739f0f9092a4a1f73052d28a598e36f0b8

**Comment by atbagga** [comment]
@qzdl Thank you for the suggested fix. We have taken the workaround for now until this gets fixed in Azure CLI. 
Feel free to raise a PR in the future for similar cases.

**Comment by atbagga** [comment]
Fix will be available in the next release 0.9.0, scheduled to release in this week.
